pkg/subctl/cmd/root.go

```diff
@@ -154,20 +153,32 @@ func addLabelsToNode(c kubernetes.Interface, nodeName string, labels map[string]
 	labelString := "{" + strings.Join(tokens, ",") + "}"
 	patch := fmt.Sprintf(`{"metadata":{"labels":%v}}`, labelString)
 
-	var err error
-
 	// retry is necessary because nodes get updated every 10 seconds, and a patch can happen
 	// in the middle of an update
-	for attempt := 0; attempt < retries; attempt++ {
-		_, err = c.CoreV1().Nodes().Patch(nodeName, types.MergePatchType, []byte(patch))
-		if err != nil {
-			if !errors.IsConflict(err) {
-				return err
+
+	var lastErr error
+	err := wait.ExponentialBackoff(nodeLabelBackoff, func() (bool, error) {
+		_, lastErr = c.CoreV1().Nodes().Patch(nodeName, types.MergePatchType, []byte(patch))
+		if lastErr != nil {
+			if !errors.IsConflict(lastErr) {
+				return false, lastErr
 			}
+			return false, nil
 		} else {
-			break
+			return true, nil
 		}
-		time.Sleep(retryInterval)
+	})
+
+	if err == wait.ErrWaitTimeout {
+		return lastErr
 	}
-	return err
+
+	return nil
```

Comment:
this should probably be "err" in this case, as it could be the "non-conflict" case, ... editing..